### DISCUSSION TITLE
Compability with Ruby 3.3

### DISCUSSION THREAD

**Discussion by tientt-holistics** [mongodb/mongo-ruby-driver#2889] [Q&A] [answered]
Hi guys, 

From the README, I can see that this gem is compatible with Ruby 3.3. But from the GitHub Actions CI, I only see that the gem is being tested on Ruby 2.7, 3.0, 3.1, and 3.2, not 3.3.
Is this gem truly compatible with Ruby 3.3?

**Comment by jamis** [comment, marked as answer]
The README jumped the gun a bit. :) We do not currently expect any problems with Ruby 3.3 compatibility (and in fact, I use Ruby 3.3 as my primary Ruby version when working on the driver), but we are still in process of adding Ruby 3.3 to our testing matrix.

The canonical place where we document compatibility with different Ruby versions is here: https://www.mongodb.com/docs/ruby-driver/current/reference/driver-compatibility/#ruby-compatibility . Once we're fully ready to announce Ruby 3.3 compatibility, we'll update that matrix accordingly.

So, loosely, yes, the gem is compatible with Ruby 3.3, but without any continuous test against that version, we can't confidently say that it's safe for use in production with Ruby 3.3. I expect we'll have that rectified in the near future.

**Comment by alexbevi** [comment]
@tientt-holistics FYI we also have [RUBY-3510](https://jira.mongodb.org/browse/RUBY-3510) open at the moment for this work, so if you watch that ticket in JIRA once it's actioned we'll have 3.3 enabled in our matrices.

**Comment by tientt-holistics** [comment]
Awesome, thank you guys for your answers.
